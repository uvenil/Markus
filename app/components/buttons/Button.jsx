'use strict';

import React from 'react';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import FontIcon from 'material-ui/FontIcon';
import Constants from '../../utils/Constants';
import _ from 'lodash';

const Button = props => {
    const icon = props.icon ?
        <FontIcon
            className={'fa fa-fw fa-' + props.icon}
            style={{ marginLeft : props.labelPosition === 'before' || !props.label ? 0 : Constants.PADDING_X1, marginRight : props.labelPosition === 'after' || !props.label ? 0 : Constants.PADDING_X1, fontSize : Constants.TEXT_FONT_SIZE }} /> : undefined;

    const style = { minWidth : props.width, height : props.height, lineHeight : props.height + 'px', textAlign : props.align };

    if (props.borderless) {
        return (
            <FlatButton
                label={props.label}
                labelPosition={props.labelPosition}
                labelStyle={props.label ? { paddingLeft : Constants.PADDING_X1, paddingRight : Constants.PADDING_X1, fontSize : Constants.TEXT_FONT_SIZE, fontWeight : props.labelWeight === 'light' ? 300 : props.labelWeight === 'bold' ? 500 : 400, textTransform : 'none' } : {}}
                icon={icon}
                primary={props.color === 'primary'}
                secondary={props.color === 'secondary'}
                disabled={props.disabled}
                style={_.assign(style, props.style)}
                onTouchTap={props.onTouchTap} />
        );
    }

    return (
        <RaisedButton
            label={props.label}
            labelStyle={props.label ? { paddingLeft : Constants.PADDING_X1, paddingRight : Constants.PADDING_X1, fontSize : Constants.TEXT_FONT_SIZE, textTransform : 'none' } : {}}
            icon={icon}
            primary={props.color === 'primary'}
            secondary={props.color === 'secondary'}
            disabled={props.disabled}
            style={_.assign(style, props.style)}
            onTouchTap={props.onTouchTap} />
    );
};

Button.propTypes = {
    label         : React.PropTypes.string,
    labelPosition : React.PropTypes.oneOf([ 'before', 'after' ]),
    labelWeight   : React.PropTypes.oneOf([ 'light', 'normal', 'bold' ]),
    icon          : React.PropTypes.string,
    borderless    : React.PropTypes.bool,
    color         : React.PropTypes.oneOf([ 'default', 'primary', 'secondary' ]),
    width         : React.PropTypes.oneOfType([ React.PropTypes.number, React.PropTypes.string ]),
    height        : React.PropTypes.oneOfType([ React.PropTypes.number, React.PropTypes.string ]),
    align         : React.PropTypes.oneOf([ 'left', 'center', 'right' ]),
    disabled      : React.PropTypes.bool,
    style         : React.PropTypes.object,
    onTouchTap    : React.PropTypes.func
};

Button.defaultProps = {
    labelPosition : 'after',
    labelWeight   : 'bold',
    width         : Constants.BUTTON_MIN_WIDTH,
    height        : Constants.BUTTON_MIN_HEIGHT,
    borderless    : true,
    color         : 'default',
    align         : 'center',
    disabled      : false
};

export default Button;
