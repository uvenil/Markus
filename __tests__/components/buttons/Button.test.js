'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import Button from '../../../app/components/buttons/Button.jsx';

it('renders correctly', () => {
    const button = Renderer.create(
        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
            <Button
                label=""
                icon="cog"
                color="primary" />
        </MuiThemeProvider>
    ).toJSON();

    expect(button).toMatchSnapshot();
});
