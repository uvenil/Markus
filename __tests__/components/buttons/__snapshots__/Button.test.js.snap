// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  disabled={false}
  href={undefined}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={[Function]}
  onTouchTap={[Function]}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
      "backgroundColor": "rgba(0, 0, 0, 0)",
      "border": 10,
      "borderRadius": 2,
      "boxSizing": "border-box",
      "color": "#00bcd4",
      "cursor": "pointer",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": "inherit",
      "fontWeight": "inherit",
      "height": 28,
      "lineHeight": "28px",
      "margin": 0,
      "minWidth": 96,
      "muiPrepared": true,
      "outline": "none",
      "overflow": "hidden",
      "padding": 0,
      "position": "relative",
      "textAlign": "center",
      "textDecoration": "none",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "userSelect": "none",
      "verticalAlign": null,
      "zIndex": 1,
    }
  }
  tabIndex={0}
  type="button"
>
  <div
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  >
    <span
      className="fa fa-fw fa-cog"
      color="#00bcd4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "#00bcd4",
          "display": "inline-block",
          "fontSize": 13,
          "fontWeight": 400,
          "marginLeft": 8,
          "marginRight": 0,
          "muiPrepared": true,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": "inherit",
        }
      }
    />
    <span
      style={
        Object {
          "fontSize": 15,
          "fontWeight": 500,
          "letterSpacing": 0,
          "muiPrepared": true,
          "paddingLeft": 8,
          "paddingRight": 8,
          "position": "relative",
          "textTransform": "none",
          "verticalAlign": "inherit",
        }
      }
    >
      Hello, World!
    </span>
  </div>
</button>
`;
